File: 1218 (Longest Arithmetic Subsequence of Given Difference)/javascript/Solution.js
/**
 * @param {number[]} arr
 * @param {number} difference
 * @return {number}
 */
var longestSubsequence = function(arr, difference) {
  let res = 1;
  const dic = new Map();

  for(let i = 0;i<arr.length;i++){
      let cnt  = 1 + (dic.get(arr[i]-difference) || 0);
      res = Math.max(res,cnt);
      dic.set(arr[i], cnt);
  }
  return res;
};

File: 1218 (Longest Arithmetic Subsequence of Given Difference)/java/Solution.java
class Solution {
    public int longestSubsequence(int[] arr, int difference) {
        int res = 1;
        Map<Integer,Integer> dic = new HashMap<>();
        for(int i =0;i<arr.length;i++){
            int cnt  = dic.getOrDefault(arr[i]-difference, 0) + 1;
            res = Math.max(res,cnt);
            dic.put(arr[i],cnt );
        }
        return res;
    }
}